MathLib/LinAlg/MatrixVectorTraits.cpp

```diff
@@ -105,7 +138,16 @@ std::unique_ptr<PETScVector>
 MatrixVectorTraits<PETScVector>::
 newInstance(MatrixSpecifications const& spec)
 {
-    return std::unique_ptr<PETScVector>(new PETScVector(spec.nrows));
+    auto const is_global_size = false;
+
+    if (spec.dof_table) {
+        auto& dt = *spec.dof_table;
```

Comment:
Actually, it will be const automatically, because we dereference a `const*const`. But I can change it.